### PR TITLE
feat(dashdirect): add login function with temporary UI

### PR DESCRIPTION
**Pull request by hadia**
## Issue being fixed or feature implemented

Add sign in dialog for dashdirect Api
## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [ ] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/SearchFragment.kt [hunk 1]
```diff
@@ -500,7 +517,22 @@ class SearchFragment : Fragment(R.layout.fragment_search) {
...
+            }
+        }
+
+        viewModel.dashDirectSignIn.observe(viewLifecycleOwner) {
+            if (it) {
+                // TODO open Buy card UI
+                Toast.makeText(requireContext(), "Open Buy Card", Toast.LENGTH_SHORT).show()
+                viewModel.logEvent(AnalyticsConstants.Explore.MERCHANT_DETAILS_BUY_GIFT_CARD)
+            }
+        }
```
Comment (Syn-McJ) [inline features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/SearchFragment.kt:535]: The pattern of calling a function and observing result in `LiveData` is not great. It makes the execution flow more complicated and hard to track.

Instead, simply call a suspend function in a `lifecycleScope` and process the result.

File: features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/ExploreViewModel.kt [hunk 2]
```diff
@@ -201,6 +206,10 @@ class ExploreViewModel @Inject constructor(
     val screenState: LiveData<ScreenState>
         get() = _screenState
 
+    private val _dashDirectSignIn = MutableLiveData<Boolean>()
+    val dashDirectSignIn: LiveData<Boolean>
+        get() = _dashDirectSignIn
```
Comment (Syn-McJ) [inline features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/ExploreViewModel.kt:211]: You don't need this. See below.

File: features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/ExploreViewModel.kt [hunk 3]
```diff
@@ -390,6 +400,52 @@ class ExploreViewModel @Inject constructor(
...
 
+    fun isUserSignInDashDirect() = repository.isUserSignIn()
+
+    fun signInToDashDirect(email: String, password: String) {
+        if (repository.isUserSignIn()) {
+            _dashDirectSignIn.value = true
+            return
+        }
+
+        viewModelScope.launch {
```
Comment (Syn-McJ) [inline features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/ExploreViewModel.kt:411]: Just make `signInToDashDirect` suspend and return the result directly. See below.

**Review by Syn-McJ — CHANGES_REQUESTED** [review]
Some changes needed.

**Review by Syn-McJ — APPROVED** [review]
This is better. Some more points to improve in the future:
- `runBlocking` usage is discourage. It's better to have the method that deals with settings as `suspend`. Reading from prefs is a IO operation and should be treated in an async manner.
- Returning `ResponseResource` from the viewModel method is unnecessary. Just return the result or throw an error, then `try/catch` in the caller.

**Comment by HashEngineering** [comment]
The base branch is called `feature/DashDirect` which doesn't follow our `feature-*` nameing scheme which is required to activate CI.

**Comment by HashEngineering** [comment]
new branch created named `feature-dashdirect`

**Review by HashEngineering — APPROVED** [review]

**Pull request merged**